subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -219,12 +226,15 @@ static void reset_adv_link(void)
 	 * check.
 	 */
 	(void)k_work_cancel_delayable(&link.prot_timer);
+	(void)k_work_cancel_delayable(&link.rx.transmit_delay);
 
 	if (atomic_test_bit(link.flags, ADV_PROVISIONER)) {
-		/* Clear everything except the retransmit and protocol timer
-		 * delayed work objects.
+		/* Clear everything except the transmit_delay, retransmit and
```

Comment:
This is not clear to me. Why do we need to keep untouched these fields if link is reset?

Reply:
We can not clear the delayable timers as they are only initialized in the init function. If they are cleared the work scheduler will crash. But the delayed timers is canceled in some lines above.